R/summary_funs.R

```diff
@@ -406,92 +438,120 @@ get_stat <- function(mu, lppd, y_wobs_test, stat, mu.bs = NULL, lppd.bs = NULL,
         }
-        crrct.bs <- mu.bs == y
+        correct_baseline <- mu_baseline == y
+      } else {
+        correct_baseline <- 0
+      }
 
-        value <- mean(wcv * (crrct - crrct.bs), na.rm = TRUE)
-        value.se <- weighted.sd(crrct - crrct.bs, wcv, na.rm = TRUE) /
-          sqrt(n_notna)
+      if (!is.null(summaries_fast) && sum(n_loo<n)) {
+        # subsampling difference estimator (Magnusson et al., 2020)
+        mu_fast <- summaries_fast$mu
+        if (!is.factor(mu_fast)) {
+          mu_fast <- round(mu_fast)
+        }
+        correct_fast <- mu_fast == y
+        srs_diffe <- .srs_diff_est_w(y_approx = correct_fast - correct_baseline,
+                                     y = (correct-correct_baseline)[loo_inds],
+                                     y_idx = loo_inds,
+                                     w = wcv)
+        value <- srs_diffe$y_hat / n + mean(wcv * correct_baseline)
+        # combine estimates of var(y_hat) and var(y)
+        value_se <- sqrt(srs_diffe$v_y_hat + srs_diffe$hat_v_y) / n
       } else {
-        value <- mean(wcv * crrct, na.rm = TRUE)
-        value.se <- weighted.sd(crrct, wcv, na.rm = TRUE) / sqrt(n_notna)
+        # full LOO estimator
+        value <- mean(wcv * correct)
+        value_se <- .weighted_sd(correct - correct_baseline, wcv) / sqrt(n)
       }
     } else if (stat == "auc") {
-      if (!is.null(mu.bs)) {
-        auc.data <- cbind(y, mu, wcv)
-        auc.data.bs <- cbind(y, mu.bs, wcv)
-        value <- auc(auc.data) - auc(auc.data.bs)
-        idxs_cols <- seq_len(ncol(auc.data))
-        idxs_cols_bs <- setdiff(seq_len(ncol(auc.data) + ncol(auc.data.bs)),
+      if (!is.null(summaries_fast) && sum(n_loo<n)) {
+        # subsampling LOO with AUC not implemented. Using fast LOO result.
+        mu <- summaries_fast$mu
+      }
+      if (!is.null(mu_baseline)) {
+        auc_data <- cbind(y, mu, wcv)
+        auc_data_baseline <- cbind(y, mu_baseline, wcv)
+        value <- .auc(auc_data)
+        idxs_cols <- seq_len(ncol(auc_data))
+        idxs_cols_bs <- setdiff(seq_len(ncol(auc_data) + ncol(auc_data_baseline)),
                                 idxs_cols)
         diffvalue.bootstrap <- bootstrap(
-          cbind(auc.data, auc.data.bs),
+          cbind(auc_data, auc_data_baseline),
           function(x) {
-            auc(x[, idxs_cols, drop = FALSE]) -
-              auc(x[, idxs_cols_bs, drop = FALSE])
+            .auc(x[, idxs_cols, drop = FALSE]) -
+              .auc(x[, idxs_cols_bs, drop = FALSE])
           },
           ...
         )
-        value.se <- sd(diffvalue.bootstrap, na.rm = TRUE)
+        value_se <- sd(diffvalue.bootstrap, na.rm = TRUE)
         lq_uq <- quantile(diffvalue.bootstrap,
                           probs = c(alpha_half, one_minus_alpha_half),
-                          names = FALSE, na.rm = TRUE)
+                          names = FALSE, na.rm = TRUE) +
+          .auc(auc_data_baseline)
```

Comment:
Previously, we did not add the baseline model AUC here. Why was this addition added?

Reply:
To use original scale for auc even when deltas=TRUE

Reply:
Ok, then this part of the code is correct, under the assumption that you want to keep the new `deltas = TRUE` behavior.